src/core/net/tcp6.hpp

```diff
@@ -0,0 +1,302 @@
+
+    /**
+     * See documentation for otTcpEndpointGetInstance in include/openthread/tcp.h.
+     */
+    static Instance *GetInstance(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpEndpointGetContext in include/openthread/tcp.h.
+     */
+    static void *GetContext(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpGetLocalAddress in include/openthread/tcp.h.
+     */
+    static const SockAddr &GetLocalAddress(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpGetPeerAddress in include/openthread/tcp.h.
+     */
+    static const SockAddr &GetPeerAddress(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpBind in include/openthread/tcp.h.
+     */
+    Error Bind(Endpoint &aEndpoint, const SockAddr &aSockName);
+
+    /**
+     * See documentation for otTcpConnect in include/openthread/tcp.h.
+     */
+    Error Connect(Endpoint &aEndpoint, const SockAddr &aSockName, uint32_t aFlags);
+
+    /**
+     * See documentation for otTcpSendByReference in include/openthread/tcp.h.
+     */
+    Error SendByReference(Endpoint &aEndpoint, otLinkedBuffer &aBuffer, uint32_t aFlags);
+
+    /**
+     * See documentation for otTcpSendByExtension in include/openthread/tcp.h.
+     */
+    Error SendByExtension(Endpoint &aEndpoint, size_t aNumBytes, uint32_t aFlags);
+
+    /**
+     * See documentation for otTcpReceiveByReference in include/openthread/tcp.h.
+     */
+    Error ReceiveByReference(Endpoint &aEndpoint, const otLinkedBuffer *&aBuffer);
+
+    /**
+     * See documentation for otTcpReceiveContiguify in include/openthread/tcp.h.
+     */
+    Error ReceiveContiguify(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpCommitReceive in include/openthread/tcp.h.
+     */
+    Error CommitReceive(Endpoint &aEndpoint, size_t aNumBytes, uint32_t aFlags);
+
+    /**
+     * See documentation for otTcpSendEndOfStream in include/openthread/tcp.h.
+     */
+    Error SendEndOfStream(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpAbort in include/openthread/tcp.h.
+     */
+    Error Abort(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpEndpointDeinitialize in include/openthread/tcp.h.
+     */
+    Error Deinitialize(Endpoint &aEndpoint);
+
+    /**
+     * See documentation for otTcpListenerInitialize in include/openthread/tcp.h.
+     */
+    Error Initialize(Listener &aListener, otTcpListenerInitializeArgs &aArgs);
+
+    /**
+     * See documentation for otTcpListenerGetInstance in include/openthread/tcp.h.
+     */
+    static Instance *GetInstance(Listener &aListener);
+
+    /**
+     * See documentation for otTcpListenerGetContext in include/openthread/tcp.h.
+     */
+    static void *GetContext(Listener &aListener);
+
+    /**
+     * See documentation for otTcpListen in include/openthread/tcp.h.
+     */
+    Error Listen(Listener &aListener, const SockAddr &aSockName);
```

Comment:
How about defining these as methods of `Listener` class?

Reply:
I'm happy to do this.

I was originally following the precedent set by the `Udp` class (see my response to your other comment about defining some methods in the `Endpoint` class).